pkg/notifier/notifier.go

```diff
@@ -0,0 +1,61 @@
+package notifier
+
+import (
+	"context"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/go-logr/logr"
+)
+
+type Notifier interface {
+	Notify(context.Context, string) error
+}
+
+type SNSNotifier struct {
+	SnsClient *sns.Client
+}
+
+var logger logr.Logger
+
+/*
+	To create the client, make sure you set the
+	AWS credentials as environment variables:
+	AWS_ACCESS_KEY_ID,
+	AWS_SECRET_ACCESS_KEY,
+	AWS_SESSION_TOKEN (optional)
+	In addition, make sure you also set the
+	NOTIFICATION_REGION for the region,
+	NOTIFICATION_TOPIC_ARN for the SNS topic ARN
+*/
+
+func NewSNSNotifier(ctx context.Context) (*SNSNotifier, error) {
+	logger := logger.WithName("Notifier")
+	region := os.Getenv("NOTIFICATION_REGION")
+	sdkConfig, err := config.LoadDefaultConfig(ctx, config.WithRegion(region))
+	if err != nil {
+		logger.Error(err, "Couldn't load default configuration. Have you set up your AWS account?")
+		return &SNSNotifier{}, err
+	}
+
+	snsNotifer := SNSNotifier{
+		SnsClient: sns.NewFromConfig(sdkConfig),
+	}
+
+	return &snsNotifer, nil
+}
+
+func (s *SNSNotifier) Notify(ctx context.Context, message string) error {
```

Comment:
We should test this function as well.

Reply:
We should add unit tests to this file.
For now I only added the Notify test to the integration tests, I will work on the tests on another PR

Reply:
I think we're developing bad habits with moving tests to later PRs. 
is there a particular reason not to add the unit tests for this function still with this PR?